### PR TITLE
One step Login/ Sign up flow

### PR DESCRIPTION
**Pull request by sa-mao**
Enable One step login and Sign up flow:
Screenshots:
#### Login:  

Screenshot 2019-10-22 at 15 01 49

#### Sign Up:

image

### CODE REVIEW

File: example/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
         android:theme="@style/AppTheme"
         tools:ignore="AllowBackup,GoogleAppIndexingWarning">
 
-        <meta-data android:name="@string/schacc_conf_locale" android:value="nb_NO" />
+        <meta-data android:name="@string/schacc_conf_locale" android:value="en_EN" />
```
[the remark this answers is not in the document]
Comment (sa-mao) [reply example/src/main/AndroidManifest.xml]: Not necessary it was just more convenient while testing :D

File: ui/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
     <application>
         <activity android:name="com.schibsted.account.ui.login.flow.passwordless.PasswordlessActivity" android:launchMode="singleTop" />
-        <activity android:name="com.schibsted.account.ui.login.flow.password.PasswordActivity" android:launchMode="singleTop">
+        <activity android:name="com.schibsted.account.ui.login.flow.password.PasswordActivity" android:launchMode="singleTop" android:windowSoftInputMode="adjustNothing">
```
[the remark this answers is not in the document]
Comment (sa-mao) [reply ui/src/main/AndroidManifest.xml]: I had to add us otherwise the view shrunk horizontally. My wish for it would be to move up a bit so we don't have fields hidden by the keyboard.

image

File: ui/src/main/res/layout-land/schacc_one_step_signup_fragment_layout.xml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        app:layout_constraintEnd_toEndOf="@+id/logos_container"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/logos_container"
+        tools:text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce malesuada tristique dolor, vel lacinia justo semper eget. Curabitur non urna non felis tincidunt faucibus at non dui."
+        tools:visibility="visible" />
+
+    <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/schacc_rounded_container"
```
[the remark this answers is not in the document]
Comment (sa-mao) [reply ui/src/main/res/layout-land/schacc_one_step_signup_fragment_layout.xml]: Why not?

File: build.gradle [hunk 4]
```diff
@@ -58,13 +58,13 @@ ext {
...
-            support         : "27.1.1"
+            support         : "28.0.0"
     ]
 
     conf = [
             minSdk    : 14,
-            targetSdk : 27,
-            compileSdk: 27,
+            targetSdk : 28,
+            compileSdk: 28,
```
[the remark this answers is not in the document]
Comment (sa-mao) [reply build.gradle]: This PR actually depends on this change.

[the remark this answers is not in the document]
Comment (zamzterz) [reply build.gradle]: Then it could be split out into a separate PR and we can merge that before this one.

File: ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+            if (actionId == EditorInfo.IME_ACTION_NEXT) {
+                identifyUser(inputFieldView, TrackingData.Screen.ONE_STEP_LOGIN) {}
+            }
+            false
+        }
+
+        credInputFieldView.setImeAction(EditorInfo.IME_ACTION_DONE) { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                identifyUser(inputFieldView, TrackingData.Screen.ONE_STEP_LOGIN) {
+                    signUpUser()
```
Comment (zamzterz) [inline ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt]: Should this be `signInUser()`?

File: ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+                }
+            }
+            false
+        }
+
+        primaryActionView.setOnClickListener {
+            identifyUser(inputFieldView, TrackingData.Screen.ONE_STEP_LOGIN) {
+                signUpUser()
```
Comment (zamzterz) [inline ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt]: Should this be `signInUser()`?

File: ui/src/main/java/com/schibsted/account/ui/login/BaseLoginActivity.kt [hunk 7]
```diff
@@ -220,6 +220,10 @@ abstract class BaseLoginActivity : AppCompatActivity(), NavigationListener {
                     setResult(AccountUi.RESULT_ERROR, Intent().putExtra(AccountUi.EXTRA_ERROR, result.error))
                     finish()
                 }
+
+                else -> {
+                    navigateToIdentificationFragment(null, viewModel, idProvider)
```
Comment (zamzterz) [inline ui/src/main/java/com/schibsted/account/ui/login/BaseLoginActivity.kt]: When does this case occur (i.e. how do I test/reproduce it)?

File: ui/src/main/java/com/schibsted/account/ui/login/BaseLoginActivity.kt [hunk 8]
```diff
@@ -323,7 +327,7 @@ abstract class BaseLoginActivity : AppCompatActivity(), NavigationListener {
                     provider = idProvider,
                     flowType = flowType,
                     flowSelectionListener = flowSelectionListener,
-                    clientInfo = clientInfo)
+                    clientInfo = clientInfo!!)
```
Comment (zamzterz) [inline ui/src/main/java/com/schibsted/account/ui/login/BaseLoginActivity.kt]: This should not be done, given it's explicitly called with `null` above it would result in an exception.

File: ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt [hunk 9]
```diff
@@ -269,7 +275,6 @@ class OneStepLoginFragment : FlowFragment<OneStepLoginContract.Presenter>(), One
 
     private fun showSignIn() {
         isSignup = false
-        BaseLoginActivity.tracker?.eventInteraction(TrackingData.InteractionType.SEND, TrackingData.Screen.ONE_STEP_LOGIN)
```
Comment (zamzterz) [inline ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt]: Why was these removed, are they sent elsewhere?

File: ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt [hunk 10]
```diff
@@ -317,11 +322,13 @@ class OneStepLoginFragment : FlowFragment<OneStepLoginContract.Presenter>(), One
          * @param uiConfiguration
          * @return a parametrized instance of [MobileIdentificationFragment]
          */
-        fun newInstance(uiConfiguration: InternalUiConfiguration, clientInfo: ClientInfo): OneStepLoginFragment {
+        fun newInstance(uiConfiguration: InternalUiConfiguration, clientInfo: ClientInfo?): OneStepLoginFragment {
             val args = Bundle()
             val fragment = OneStepLoginFragment()
             args.putParcelable(KEY_UI_CONF, uiConfiguration)
-            args.putParcelable(KEY_CLIENT_INFO, clientInfo)
+            if (clientInfo != null) {
```
Comment (zamzterz) [inline ui/src/main/java/com/schibsted/account/ui/login/screen/onesteplogin/OneStepLoginFragment.kt]: Could be more idiomatic with:
```kotlin
clientInfo.also {
  args.putParcelable(KEY_CLIENT_INFO, it)
}
```
